docs/en/changelog/migration_2_5.rst

```diff
@@ -0,0 +1,690 @@
+
+Behavioral Changes (BC Breaks)
+------------------------------
+
+NamingStrategy interface changed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``Doctrine\ORM\Mapping\NamingStrategyInterface`` changed slightly
+to pass the Class Name of the entity into the join column name generation:
+
+:: 
+
+    -    function joinColumnName($propertyName);
+    +    function joinColumnName($propertyName, $className = null);
+
+It also received a new method for supporting embeddables:
+
+::
+
+    public function embeddedFieldToColumnName($propertyName, $embeddedColumnName);
+
+Minor BC BREAK: EntityManagerInterface instead of EntityManager in type-hints
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ 
+As of 2.5, classes requiring the ``EntityManager`` in any method signature will now require 
+an ``EntityManagerInterface`` instead.
+If you are extending any of the following classes, then you need to check following
+signatures:
+
+- ``Doctrine\ORM\Tools\DebugUnitOfWorkListener#dumpIdentityMap(EntityManagerInterface $em)``
+- ``Doctrine\ORM\Mapping\ClassMetadataFactory#setEntityManager(EntityManagerInterface $em)``
+
+Minor BC BREAK: Custom Hydrators API change
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As of 2.5, `AbstractHydrator` does not enforce the usage of cache as part of
+API, and now provides you a clean API for column information through the method
+`hydrateColumnInfo($column)`.
+Cache variable being passed around by reference is no longer needed since
+Hydrators are per query instantiated since Doctrine 2.4.
+
+- `DDC-3060 <http://doctrine-project.org/jira/browse/DDC-3060>`_
+
+Minor BC BREAK: Entity based EntityManager#clear() calls follow cascade detach
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Whenever ``EntityManager#clear()`` method gets called with a given entity class
+name, until 2.4, it was only detaching the specific requested entity.
+As of 2.5, ``EntityManager`` will follow configured cascades, providing a better
+memory management since associations will be garbage collected, optimizing
+resources consumption on long running jobs.
+
+Updates on entities scheduled for deletion are no longer processed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Doctrine 2.4, if you modified properties of an entity scheduled for deletion, UnitOfWork would
+produce an UPDATE statement to be executed right before the DELETE statement. The entity in question
+was therefore present in ``UnitOfWork#entityUpdates``, which means that ``preUpdate`` and ``postUpdate``
+listeners were (quite pointlessly) called. In ``preFlush`` listeners, it used to be possible to undo
+the scheduled deletion for updated entities (by calling ``persist()`` if the entity was found in both
+``entityUpdates`` and ``entityDeletions``). This does not work any longer, because the entire changeset
+calculation logic is optimized away.
```

Comment:
is it entirely impossible to cancel the deletion in the listener now ? This would break the soft-deleteable extension which is relying on that.